crates/solver/src/settlement_access_list.rs

```diff
@@ -151,7 +213,63 @@ impl AccessListEstimating for TenderlyApi {
                 .map(Into::into)
                 .collect())
         }))
-        .await
+        .await)
+    }
+}
+
+fn resolve_call_request(tx: &TransactionBuilder<DynTransport>) -> Result<(H160, H160, Bytes)> {
+    let from = tx
+        .from
+        .clone()
+        .context("transaction from does not exist")?
+        .address();
+    let to = tx.to.context("transaction to does not exist")?;
+    let data = tx.data.clone().context("transaction data does not exist")?;
+    Ok((from, to, data))
+}
+
+// this function should remove duplicates and elements that are not useful
+// currently only eliminating addresses and storages with value '1'
+// should be updated continually as we learn more about the imperfections of 3rd party access_list calculators
+#[allow(dead_code)]
+fn filter_access_list(access_list: AccessList) -> AccessList {
+    access_list
+        .into_iter()
+        .filter(|item| {
+            item.address != H160::from_low_u64_be(1)
+                && item
+                    .storage_keys
+                    .iter()
+                    .all(|key| key != &H256::from_low_u64_be(1))
+        })
+        .collect()
+}
+
+pub struct PriorityAccessListEstimating {
+    estimators: Vec<Box<dyn AccessListEstimating>>,
+}
+
+impl PriorityAccessListEstimating {
+    pub fn new(estimators: Vec<Box<dyn AccessListEstimating>>) -> Self {
+        Self { estimators }
+    }
+}
+
+#[async_trait::async_trait]
+impl AccessListEstimating for PriorityAccessListEstimating {
+    async fn estimate_access_lists(
+        &self,
+        txs: &[TransactionBuilder<DynTransport>],
+    ) -> Result<Vec<Result<AccessList>>> {
+        for (i, estimator) in self.estimators.iter().enumerate() {
+            match estimator.estimate_access_lists(txs).await {
```

Comment:
If latency is a big concern the access lists could be estimated in parallel here with `futures::future::select_all()`. This will only give us anything if the high priority estimator fails. Since this would make the implementation non-trivial it might be worth it to measure first how often this scenario happens.

Reply:
I expect the high priority estimator to be quite reliable (and much faster than the second one). But I will remember this comment if improvements are needed.